contracts/NFTVesting/IntervalVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,169 @@
+    }
+
+    /**
+     * @dev Returns the smart contract NFT.
+     */
+    function nft() public view virtual returns (IERC721) {
+        return _nft;
+    }
+
+    /**
+     * @dev Returns the token ID of the NFT being held.
+     */
+    function tokenId() public view virtual returns (uint256) {
+        return _tokenId;
+    }
+
+    /**
+     * @dev Returns the beneficiary that will receive the NFT.
+     */
+    function beneficiary() public view virtual returns (address) {
+        return _beneficiary;
+    }
+
+    /**
+     * @dev Returns the time when the NFT are released in seconds since Unix Interval (i.e. Unix timestamp).
+     */
+    function releaseTime() public view virtual returns (uint256) {
+        return _releaseTime;
+    }
+
+    /**
+     * @dev Returns discount for locking at each Interval
+     */
+    function discountPerInterval() public view virtual returns (uint256) {
+        return _maxDiscount / _maxInterval;
+    }
+
+    /**
+     * @dev Returns current Interval
+     */
+    function getCurrentInterval() public view returns (uint256) {
+        if (block.timestamp < _releaseTime) {
+            return 0;
+        }
+        uint256 interval = uint256(block.timestamp - _releaseTime) /
+            _intervalDuration;
+        if (interval > _maxInterval) {
+            interval = _maxInterval;
+        }
+        return interval;
+    }
+
+    /**
+     * @dev Returns the remaining interval before max vesting interval
+     */
+    function getIntervalsLeft() public view returns (uint256) {
+        return _maxInterval - _currentInterval;
+    }
+
+    /**
+     * @dev Updates the interval after release schedule
+     */
+    function updateInterval() private validRelease {
+        require(_currentInterval <= _maxInterval, "Vesting Schedule is over.");
+        // floor the current Interval to the nearest Interval interval
+        _currentInterval = getCurrentInterval();
+    }
+
+    /**
+     * @dev Get the current discount in terms of percentage for the vesting schedule.
+     */
+    function getDiscount() public view returns (uint256) {
+        require(_currentInterval > 0, "Vesting Schedule is not Up yet.");
+        return (_currentInterval * discountPerInterval()) / 100;
+    }
+
+    /**
+     * @dev Transfers NFT held by the timelock to the beneficiary. Will only succeed if invoked after the release
+     * time. Sends the discount in Eth to the beneficiary.
+     */
+
+    function release() public virtual validRelease {
+        require(
+            nft().ownerOf(tokenId()) == address(this),
+            "TimeLock: no NFT to release for user"
+        );
+
+        updateInterval();
+        uint256 discount = getDiscount();
+        uint256 ethDiscount = discount * address(this).balance;
+        (bool sent, ) = beneficiary().call{value: ethDiscount}("");
+        require(sent, "Failed to send Ether");
+
+        nft().safeTransferFrom(address(this), beneficiary(), tokenId());
```

Comment:
Same on ability to make atomic

Reply:
sorry which part can i reduce to make this function more atomic